### PR TITLE
Update function call expr source gen and fix few issues related to function call exprs

### PR DESCRIPTION
**Pull request by LakshanWeerasinghe**
## Purpose
$subject

### CODE REVIEW

File: flow-model-generator/modules/flow-model-generator-ls-extension/src/test/resources/available_nodes/config/connector1.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
           "items": [
             {
               "metadata": {
-                "label": "post:[/path/to/resource]",
+                "label": "post:/path/to/subdirectory",
```
Comment (hasithaa) [inline flow-model-generator/modules/flow-model-generator-ls-extension/src/test/resources/available_nodes/config/connector1.json:22]: subdirectory ?

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CommonUtils.java [hunk 2]
```diff
@@ -507,12 +501,10 @@ public static String buildResourcePathTemplate(FunctionSymbol functionSymbol) {
...
                 }
                 ((PathSegmentList) resourcePath).pathRestParameter().ifPresent(pathRestParameter -> {
-                    String type = CommonUtil.getRawType(pathRestParameter.typeDescriptor())
-                            .signature();
-                    pathBuilder.append("[").append(type).append("...]");
+                    pathBuilder.append("/path/to/subdirectory");
                 });
             }
-            case PATH_REST_PARAM -> pathBuilder.append("[").append("/path/to/resource").append("]");
+            case PATH_REST_PARAM -> pathBuilder.append("/path/to/subdirectory");
```
Comment (hasithaa) [inline flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CommonUtils.java:507]: why subdirectory ?

Comment (LakshanWeerasinghe) [reply flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CommonUtils.java:507]: resource is an escaped keyword. In url conventions they call it sub directory

Comment (hasithaa) [reply flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CommonUtils.java:507]: OK. That makes sense.

**Review by hasithaa — APPROVED** [review]

**Pull request merged**